### DISCUSSION TITLE
layout: breaks upon refreshing

### DISCUSSION THREAD

**Discussion by vedxp** [vuejs/vitepress#1539] [General]
#### Hey guys, I'm facing this issue:

- Deployed my VitePress site on Vercel 
- Visited the site 
- Navigated to any page using `doc` layout
- Refreshed the page 
- The layout broke for **every** page upon refreshing, including the ones using `home` layout

But, upon refresh a page using `home` layout, it doesn't break anything.

Here's a screenshot of a page using `doc` layout:
<details><summary>Before refreshing</summary>

![Screenshot_20221026-153217~2](https://user-images.githubusercontent.com/87353286/198000178-43738069-00c3-424f-8ade-a842a99495e9.png)

</details><details><summary>After refreshing</summary>

![Screenshot_20221026-153223~2](https://user-images.githubusercontent.com/87353286/198000234-3ca38458-a934-44d0-83d6-a82d84520525.png)

</details>

#### Link to the site: [Reesors](https://reesors.vercel.app/)

---

#### I wonder if this is something I did wrong or a VitePress side issue.

> **Note**
> This issue doesn't occur if deployed on Netlify

#### Help world be really appreciated :)

**Comment by brc-dd** [comment]
Can you share your .vitepress/config.ts and vercel.json if present? Looks like you've clean urls option present on Vercel but not on VitePress (or vice versa). (`.html` is getting stripped from urls, causing hydration mismatches)

**Reply by vedxp** [reply]
> Can you share your .vitepress/config.ts and vercel.json if present? Looks like you've clean urls option present on Vercel but not on VitePress (or vice versa). (`.html` is getting stripped from urls, causing hydration mismatches)

Here's:
#### > [config.ts](https://github.com/VedxP/reesors/blob/main/docs/.vitepress/config.ts) 
#### > Build Settings set in Vercel
 - Build Command: `pnpm docs:build`
-  Output Directory: `docs/.vitepress/dist`
- Install Command: `pnpm install`

**Reply by brc-dd** [reply]
Try creating a file named `vercel.json` at top of your repo having this:

```json
{
  "cleanUrls": true
}
```

**Reply by vedxp** [reply]
> Try creating a file named `vercel.json` at top of your repo having this:
> 
> ```json
> {
>   "cleanUrls": true
> }
> ```

Yeah, thanks!
It worked like a charm.

Please consider documenting this under **Deploying on Vercel** :)

**Reply by sametcn99** [reply]
> Try creating a file named `vercel.json` at top of your repo having this:
> 
> ```json
> {
>   "cleanUrls": true
> }
> ```

It worked for me too, thanks.
